### DISCUSSION TITLE
Specify Save Path of Compiled pdf File?

### DISCUSSION THREAD

**Discussion by davidleejy** [nvarner/typst-lsp#420] [Q&A]
At the moment, the compiled pdf file is saved in the same directory as the typ file. 

1. Is it possible to specify the location to save the compiled pdf file?

2. Is it possible to specify the filename of the compiled pdf file?

I am trying to get the pdf to be saved in the parent directory containing the directory that contains the typ file as depicted in this file structure:

```
parent-dir/
├─ my-book.pdf
├─ book/
│  ├─ my-book.typ
│  ├─ template.typ
│  ├─ theorems.typ
│  ├─ math-scripts.typ
```

**Comment by astrale-sharp** [comment]
I don't think it's currently possible :thinking: 
What would the UI look like?

**Comment by davidleejy** [comment]
Hmm.. good question... I haven't really got solid thoughts for this but a couple of possibilities:

1. Each project can have its own config file that specifies such preferences. This is similar to the concept of a project's directory  containing a file with project-specific preferences for an IDE's configuration, compilation instructions, or git.
2. A single Typst LSP config file that specifies preferences for each project.

Possibility (2) appears to be less convenient should the need to share project-specific preferences among co-authors arise.

 I'm thinking of submitting an issue ticket for a feature request and perhaps we can discuss it there?
